pkg/tfbridge/provider.go

```diff
@@ -678,6 +679,39 @@ func (p *Provider) Check(ctx context.Context, req *pulumirpc.CheckRequest) (*pul
 	return &pulumirpc.CheckResponse{Inputs: minputs, Failures: failures}, nil
 }
 
+// For properties with MaxItemsOne, where the state is still an array
+// (i.e. from a previous version without MaxItemsOne)
+// we need to mark them for update manually in order to correct the state
+// from an array to a flat type.
+func markWronglyTypedStateDiff(res Resource, olds resource.PropertyMap, diff *shim.InstanceDiff) error {
+	if diff == nil {
+		return fmt.Errorf("diff is nil")
+	}
+	detectMismatchingType := func(localSchema shim.Schema, localInfo *SchemaInfo, p resource.PropertyValue) bool {
+		if localInfo != nil &&
+			localInfo.MaxItemsOne != nil &&
+			*localInfo.MaxItemsOne &&
+			localInfo.Elem == nil &&
+			p.IsArray() {
+			return true
+		}
+		return false
+	}
+	tr := func(pulumiPath resource.PropertyPath, localValue resource.PropertyValue) (resource.PropertyValue, error) {
+		schemaPath := PropertyPathToSchemaPath(pulumiPath, res.TF.Schema(), res.Schema.Fields)
+		schema, info, err := LookupSchemas(schemaPath, res.TF.Schema(), res.Schema.Fields)
```

Comment:
I think that this function `tr` should never return err. Here LookupSchemas may return err!=nil If it failed to find a schema for this path. We should not error out though in this case, because in some cases we might be handling data without a proper schema for it, no reason to give up. Instead, what's the right decision when no schema information is present here? Possibly don't do anything and just `return localValue`.